lib/src/accordion/accordion.component.ts

```diff
@@ -0,0 +1,175 @@
+import {
+    Component,
+    Input,
+    ElementRef,
+    ViewChildren,
+    QueryList,
+    OnInit,
+    OnChanges,
+    SimpleChanges,
+    ViewEncapsulation,
+    AfterViewInit,
+    ChangeDetectorRef,
+    AfterViewChecked,
+} from "@angular/core";
+import { randomId } from "@sebgroup/frontend-tools/dist/randomId";
+
+export type AccordionIconRotation = "deg-180" | "deg-180-counter" | "deg-90" | "deg-90-counter";
+export type AccordionContentType = AccordionContent | Array<AccordionContent> | string;
+export type AccordionContent = { title?: string; desc?: string };
+
+export interface AccrodionListItem {
+    header: string;
+    subHeaderText?: string;
+    content?: AccordionContentType;
+}
+
+export interface AccordionProps {
+    className?: string;
+    customIcon?: string;
+    customIconExpanded?: string;
+    iconPosition?: "left" | "right";
+    iconRotation?: AccordionIconRotation;
+    id?: string;
+    list: Array<AccrodionListItem>;
+    alternative?: boolean;
+    activeIndex?: number;
+}
+
+@Component({
+    selector: "sebng-accordion",
+    templateUrl: "./accordion.component.html",
+    styleUrls: ["./accordion.component.scss"],
+    encapsulation: ViewEncapsulation.None,
+})
+export class AccordionComponent implements OnInit, OnChanges, AfterViewInit, AfterViewChecked {
+    @Input() className?: string;
+    @Input() customIcon?: string;
+    @Input() customIconExpanded?: string;
+    @Input() iconPosition?: "left" | "right";
+    @Input() iconRotation?: AccordionIconRotation;
+    @Input() id?: string;
+    @Input() list: Array<AccrodionListItem>;
+    @Input() alternative?: boolean;
+    @Input() activeIndex?: number;
+
+    @ViewChildren("accordionItemRefs") accordionItemRefs: QueryList<ElementRef>;
+
+    public idList: Array<string>;
+    public accordionClassName: string = "custom-accordion";
+    public itemClassName: string = "custom-accordion";
+    public active: number = null;
+
+    public heightList: Array<number>;
+
+    constructor(private changeDetector: ChangeDetectorRef) {}
+
+    //helper functions
+    constructIds() {
+        this.idList = this.list.map(() => randomId("accordion-"));
+    }
+
+    /** Constructs the `className` to be used in accordion wrapper */
+    constructClassName(): void {
+        let cn: string = "custom-accordion";
+        cn += this.className ? ` ${this.className}` : "";
+        cn += this.alternative ? " alternative-accordion" : "";
+        this.accordionClassName = cn;
+    }
+
+    /** Constructs the `className` to be used in accordion items */
+    constructItemClassName(): void {
+        let cn: string = "accordion-item";
+        cn += " " + (this.iconPosition ? this.iconPosition : "left");
+        cn += " " + (this.iconRotation ? this.iconRotation : "deg-180");
+        cn += this.customIconExpanded ? " transform" : "";
+        this.itemClassName = cn;
```

Comment:
instead of this, I believe you can construct an object for `ngClass` to consume. @omarVodiaK can back me up on this

Reply:
I can actually use angular ngClass too, just got caught up with the react implementation i guess.

Reply:
Here is what it will look like if i am to go with `ngClass` . `[ngClass]="[className, { 'alternative-accordion': alternative }]"` . This however has some issue. Since both className and alternative are optional. It doesn't work and often times result in a comma after the first prop when not provided. I think having the string `itemClassName` should suffice here.

Reply:
can you move this function and the one before `constructClassName` to an `ngClass`?

Reply:
No, I couldn't because of the issue i explain above. I think constructing a string variable `itemClassName` is the right solution considering `className` and `alternative` are both optional.

Reply:
Can you at least return an ngClass object instead, I don't want us to deal with class name spaces manually
```suggestion
        return {
            "accordion-item", true,
            [this.iconPosition || "left"]: true,
            [this.iconRotation || "deg-180"]: true,
            "transform": this.customIconExpanded,
            [this.itemClassName]: this.itemClassName
        };
```

Reply:
Right now you have a structured variable within angular lifecycle that is only recalculated when necessary or one of the the props changes. Returning a function here is not the angular way and bad practice. It simply means getting recalculated when ever the ui is re-rendered or a prop change regardless. If we have to, then alternatively we can use pipe and have something like `[ngClass]=" src | calculateClass :  [iconPosition, iconRotation, customIconExpanded] "` which is the angular way. Personally i don't find that necessary.